storage/interceptable_repository.go

```diff
@@ -166,22 +167,19 @@ func (ir *interceptableRepository) Delete(ctx context.Context, objectType types.
 
 func (ir *interceptableRepository) Update(ctx context.Context, obj types.Object, labelChanges ...*query.LabelChange) (types.Object, error) {
 	objectType := obj.GetType()
-	if err := transformCredentials(ctx, obj, ir.encrypter.Encrypt); err != nil {
-		return nil, err
-	}
 
 	updateObjFunc := func(ctx context.Context, _ Repository, oldObj, newObj types.Object, labelChanges ...*query.LabelChange) (types.Object, error) {
+		if err := transformCredentials(ctx, newObj, ir.encrypter.Encrypt); err != nil {
+			return nil, err
+		}
+
 		newObj.SetUpdatedAt(time.Now().UTC())
 
 		object, err := ir.repositoryInTransaction.Update(ctx, newObj, labelChanges...)
 		if err != nil {
 			return nil, err
 		}
 
```

Comment:
To remove the overhead of an extra request - currently no post processing interceptor needs the credentials - but i can add it back (though then it will have to be added on all operations for consistency)